front/src/common/Map/Buttons/MapButtons.js

```diff
@@ -12,28 +12,71 @@ import ButtonFullscreen from 'common/ButtonFullscreen';
 import MapSearch from 'common/Map/Search/MapSearch';
 import MapSettings from 'common/Map/Settings/MapSettings';
 import MapKey from 'common/Map/MapKey';
+import { useSelector, useDispatch } from 'react-redux';
+import { updateFeatureInfoClickOSRD } from 'reducers/osrdconf';
+
+import { getFeatureInfoClick } from 'reducers/osrdconf/selectors';
+import useOutsideClick from 'utils/hooks/useOutsideClick';
 import ButtonMapInfras from './ButtonMapInfras';
 
 export default function MapButtons(props) {
   const { resetPitchBearing, withInfraButton } = props;
-  const [showSearch, setShowSearch] = useState(false);
-  const [showSettings, setShowSettings] = useState(false);
-  const [showMapKey, setShowMapKey] = useState(false);
+  const featureInfoClick = useSelector(getFeatureInfoClick);
+  const dispatch = useDispatch();
+
+  const mapModalKeywords = {
+    search: false,
+    settings: false,
+    key: false,
+  };
+
+  const [showMapModal, setShowMapModal] = useState(mapModalKeywords);
```

Comment:
I think it would be better if you keep in the state only the value of the opened modal, and not a dict like this.

You can create a dict which will contain all the possible opened modal names (in TS, you can use enums https://www.typescriptlang.org/docs/handbook/enums.html), and then create a state variable which will store the opened modal.
```
MAP_POPOVERS = { SEARCH: "SEARCH", SETTINGS: "SETTINGS", KEY: "KEY" };

const [openedPopover, setOpenedPopover] = useState(undefined);
```

Then you can show the Popover only if openedPopover has the right value
```
{ openedPopover === MAP_POPOVERS.SEARCH &&
      <MapSearch active={showMapModal.search} toggleMapSearch={() => toggleMapModal('search')} /> }
{ openedPopover === MAP_POPOVERS.SETTINGS &&
      <MapSettings
        active={showMapModal.settings}
        toggleMapSettings={() => toggleMapModal('settings')}
      /> }
{ openedPopover === MAP_POPOVERS.KEY &&
      <MapKey active={showMapModal.key} toggleMapKey={() => toggleMapModal('key')} /> }
```

Of course you will need to adapt the function toggleMapKey. Feel free to call me if you have any question

Reply:
Done